tests/test_http.py

```diff
@@ -593,6 +622,36 @@ def test_unicode(self):
     self.assertEqual(method, http.method)
     self.assertEqual(str, type(http.method))
 
+  def test_retry_ssl_errors_non_resumable(self):
+    model = JsonModel()
+    request = HttpRequest(
+        HttpMockWithSSLErrors(3, {'status': '200'}, '{"foo": "bar"}'),
+        model.response,
+        u'https://www.example.com/json_api_endpoint')
+    request._sleep = lambda _x: 0  # do nothing
+    request._rand = lambda: 10
+    response = request.execute(num_retries=3)
+    self.assertEqual({u'foo': u'bar'}, response)
+
+  def test_retry_ssl_errors_resumable(self):
+    f = open(datafile('small.png'), 'rb')
```

Comment:
Does f not need to be closed? What would you think of writing these two lines as
```
with open(datafile('small.png'), 'rb') as small_png_file:
  small_png_file_descriptor = BytesIO(small_png_file.read())
```
?

(I also happen to dislike initialisms and abbreviations for identifiers.)